### PR TITLE
Progress bar enhancement

### PR DESCRIPTION
**Pull request by iBreakEverything**
Adds more use cases for updateProgress function. Now badge page scanning and badges fetching will display a progress bar.

### CODE REVIEW

File: ASF-STM.user.js [hunk 1]
```diff
@@ -510,9 +510,9 @@
         throbber.setAttribute("style", "display: none;");
     }
 
-    function updateProgress(index) {
-        let bar = document.getElementById("asf_stm_progress");
-        let progress = 100 * ((index + 1) / bots.Result.length);
+    function updateProgress(index, total, zeroIndex=1) {
+        const bar = document.getElementById("asf_stm_progress");
+        let progress = 100 * index / (total - zeroIndex);
```
Comment (Rudokhvist) [inline ASF-STM.user.js:515]: This is outright wrong. Let's say we have index=5, total=10, zeroIndex=5. With this logic result will be 100%, while it should be 20%. I believe you should remove third parameter and only leave index and total, and rest of calculations should be in the place where you invoke this function.
Also, do you understand that you have division by zero when total=1? You are just lucky that js is shit of a language and don't consider this error, and even allows to set result as element width 🤦‍♂️

File: ASF-STM.user.js [hunk 2]
```diff
@@ -947,10 +947,11 @@
             if (userindex == -1) {
                 profileLink = myProfileLink;
                 updateMessage("Getting our data for badge " + (index + 1) + " of " + botBadges.length);
+                updateProgress(index, botBadges.length);
             } else {
                 profileLink = "profiles/" + bots.Result[userindex].SteamID;
                 updateMessage("Fetching bot " + (userindex + 1).toString() + " of " + bots.Result.length.toString() + " (badge " + (index + 1) + " of " + botBadges.length + ")");
-                updateProgress(userindex);
+                updateProgress(userindex, botBadges.length)
```
Comment (Rudokhvist) [inline ASF-STM.user.js:954]: It should be `bots.Result.length`, not `botBadges.length`. Please, at least test your script before submitting PR, it's easy to notice.

File: ASF-STM.user.js [hunk 3]
```diff
@@ -912,7 +912,7 @@
             debugPrint(new Date(Date.now()));
             hideThrobber();
             hideMessage();
-            updateProgress(bots.Result.length - 1);
+            updateProgress(bots.Result.length - 1, bots.Result.length);
```
Comment (Rudokhvist) [inline ASF-STM.user.js:915]: you changed formula in `updateProgress` but did not bother to change the `-1` here.

File: ASF-STM.user.js [hunk 4]
```diff
@@ -1164,6 +1165,7 @@
                 errors = 0;
                 debugPrint("processing page " + page);
                 updateMessage("Processing badges page " + page);
+                updateProgress(page, maxPages, 0);
```
Comment (Rudokhvist) [inline ASF-STM.user.js:1168]: you are using maxPages before it's assigned a value. Once again, you are lucky that JS is shit and don't mind.

**Review by Rudokhvist — CHANGES_REQUESTED** [review]
I don't mind showing more progress, it's a good idea, but... 
You did 5 changes and 4 of them contain errors. Please, put at least a little effort when you send a PR.

File: ASF-STM.user.js (hunk 1 above)
Comment (iBreakEverything) [reply ASF-STM.user.js:515]: zeroIndex means if the index starts from 0 or 1, as arrays start from 0 and badge pages start from 1

Comment (Rudokhvist) [reply ASF-STM.user.js:515]: So, zeroIndex=1 means that array starts from 0, and zeroIndex=0 means that array starts from 0? Excuse me, are you high?

Comment (iBreakEverything) [reply ASF-STM.user.js:515]: zeroIndex=0 means that array starts from 1

Comment (Rudokhvist) [reply ASF-STM.user.js:515]: That's some next level bullshit. I won't have such code in my script. Either make it proper or I'll implement it myself, when I'll have some time.

**Comment by iBreakEverything** [comment]
Thank you for your patience. I appreciate your time in providing this feedback and so I have updated the code based on it. I hope there are no problems this time, otherwise let me know when you have the time, so I can fix them.

**Review by Rudokhvist — APPROVED** [review]
Now this looks MUCH better! Thank you for your contribution! 🏆

**Pull request merged**